jmx/plugins/org.jboss.tools.jmx.core/src/org/jboss/tools/jmx/core/tree/NodeUtils.java

```diff
@@ -52,17 +52,19 @@ public static Root createObjectNameTree(final IConnectionWrapper connectionWrapp
 			public void run(MBeanServerConnection connection) throws Exception {
 		        monitor.beginTask(JMXCoreMessages.LoadMBeans, 1000);
 				Set beanInfo = connection.queryNames(new ObjectName("*:*"), null); //$NON-NLS-1$
-				monitor.worked(100);
-				SubProgressMonitor subMon = new SubProgressMonitor(monitor, 900);
-				subMon.beginTask(JMXCoreMessages.InspectMBeans, beanInfo.size() * 100);
 		        _root[0] = NodeBuilder.createRoot(connectionWrapper);
```

Comment:
JMX loading was unable to be canceled. Also in some cases mbeaninfo might be null (bad connection) so no reason to keep trying if the mbeaninfo cannot be loaded.